README.md

```diff
@@ -37,50 +44,108 @@ expBackoffBase = 2
 
-> Task :run
-[main] INFO spiffe.api.svid.examples.ClientExample - Fetching the SVIDs asynchronously
-[main] INFO spiffe.api.svid.examples.ClientExample - Waiting for certificates...
-[main] INFO spiffe.api.svid.examples.ClientExample - Doing other work...
-[main] INFO spiffe.api.svid.examples.ClientExample - Doing other work...
-[main] INFO spiffe.api.svid.examples.ClientExample - Doing other work...
-[grpc-default-executor-0] INFO spiffe.api.svid.examples.ClientExample - Spiffe ID fetched: spiffe://example.org/workload
-[main] INFO spiffe.api.svid.examples.ClientExample - Exiting...
+#### Register the SPIFFE Provider
 
-BUILD SUCCESSFUL in 1s
+You can extend and override the master security properties file. 
+
+Create a file `java.security` with the following content:
 
 ```
+security.provider.<n>=spiffe.provider.SpiffeProvider
 
-### Generating the JAR  
+# Determines the default key and trust manager factory algorithms for
+# the javax.net.ssl package.
+ssl.KeyManagerFactory.algorithm=Spiffe
+ssl.TrustManagerFactory.algorithm=Spiffe
 
+# The spiffeID that will be trusted
+ssl.spiffe.accept=spiffe://example.org/workload
 ```
-./gradlew build
+
+In your `java.security` file: 
+
+* replace `<n>` following the order of the `# List of Providers` in the master file. 
+
+* replace the value of the custom property `ssl.spiffe.accept` with the Spiffe ID of the workload that is allowed to connect.
+
+To pass your custom security properties file through the command line via system property when starting the JVM:
+
+```
+-Djava.security.properties=<path to java.security>
 ```
 
-The jar file `java-spiffe-0.1-SNAPSHOT.jar` is generated in folder `libs`.
+For example, it can be passed in the `JAVA_OPTS` used by the Tomcat's startup script: 
 
-### Generating the fat JAR with all the dependencies
+```
+$ export JAVA_OPTS="$JAVA_OPTS -Djava.security.properties=java.security"
+$ ./catalina.sh run
+```
+
+The properties defined in your custom properties file will override the properties in the master file. 
+
+### Configure Workload API Socket Endpoint
+
+The socket endpoint can be configured defining an environment variable named `SPIFFE_ENDPOINT_SOCKET`: 
+
+```
+export SPIFFE_ENDPOINT_SOCKET=/tmp/agent.sock
+``` 
+
+or it can be configured from the command line via system property: 
+
+```
+-Dspiffe.endpoint.socket=/tmp/agent.sock
+```
+
+If both are defined, system property overrules. 
+
+If the endpoint socket is not defined, there will be an error stating `SPIFFE_ENDPOINT_SOCKET is not defined`.
+
+### Configure a Tomcat connector
+
+A Tomcat TLS connector that uses the `Spiffe` KeyStore can be configured as follows: 
 
 ```
-./gradlew shadowJar
+<Connector
+            protocol="org.apache.coyote.http11.Http11NioProtocol"
+            port="8443" maxThreads="200"
+            scheme="https" secure="true" SSLEnabled="true"
+            keystoreFile="" keystorePass=""
+            keystoreType="Spiffe"
```

Comment:
Is it normal for these to be capitalized?

Reply:
I didn't find a naming convention for the keystoreType, but as keystoretypes are in uppercase like _PKCS12_ or _JKS_, representing acronyms, I thought it could be correct to name the type as "Spiffe" treating it as a single word and not as an acronym. I'm not sure though, maybe the best would be to name it  _"SPIFFE"_